stunnel/stunnel.ml

```diff
@@ -155,6 +155,17 @@ let pool =
   ; cert_bundle_path= "/etc/stunnel/xapi-pool-ca-bundle.pem"
   }
 
+let debug_conf () =
+  let debug_enabled =
+    (try Unix.getenv "debug_stunnel" with _ -> "") |> String.lowercase_ascii
+    |> fun x -> List.mem x ["yes"; "true"; "1"]
+  in
+  match debug_enabled with
+  | true ->
+      "debug = authpriv.7"
+  | _ ->
+      "debug = authpriv.5"
```

Comment:
I'd prefer if callers didn't have to rely on environment variables. We could expose two functions:

```suggestion
let debug_conf_of_bool verbose : string = if verbose then "debug=authpriv.7" else "debug=authpriv.5"

let debug_conf_of_env () : string =
    (try Unix.getenv "debug_stunnel" with _ -> "") |> 
    String.lowercase_ascii |> 
    fun x -> List.mem x ["yes"; "true"; "1"] |>
    debug_conf_of_bool
```

Please can you also document `extended_diagnosis`, as it's not obvious what the purpose of that is now